packages/sdk-react/src/core/queries/useCounterpart.ts

```diff
@@ -128,251 +95,145 @@ export const counterpartQueryKeys = {
-        );
-      }
-
-      return monite.api.counterpartsAddresses.getCounterpartAddressesById(
-        addressId,
-        counterpartId
-      );
+  return api.counterparts.getCounterpartsIdAddresses.useQuery(
+    {
+      path: { counterpart_id: counterpartId ?? '' },
     },
-
-    enabled: Boolean(counterpartId) && Boolean(addressId),
-  });
+    {
+      enabled: !!counterpartId,
+    }
+  );
 };
 
-export const useUpdateCounterpartAddress = (
-  addressId: string,
-  counterpartId: string
-) => {
+export const useUpdateCounterpartAddress = () => {
   const { i18n } = useLingui();
-  const { monite } = useMoniteContext();
-  const { update } = useCounterpartAddressListCache(counterpartId);
-
-  return useMutation<
-    CounterpartAddressResponseWithCounterpartID,
-    ApiError,
-    CounterpartUpdateAddress
-  >({
-    mutationFn: (payload) =>
-      monite.api.counterpartsAddresses.updateCounterpartsAddress(
-        addressId,
-        counterpartId,
-        payload
-      ),
-
-    onSuccess: (address) => {
-      update(address);
-
-      toast.success(t(i18n)`Successful update.`);
-    },
-
-    onError: () => {
-      toast.error(t(i18n)`Failed to update.`);
-    },
-  });
-};
-
-export const useCounterpartBankList = (counterpartId?: string) => {
-  const { monite } = useMoniteContext();
-
-  return useQuery<CounterpartBankAccountResponse[], Error>({
-    queryKey: counterpartQueryKeys.bankList(counterpartId),
-
-    queryFn: () =>
-      !!counterpartId
-        ? monite.api.counterparts
-            .getBankAccounts(counterpartId)
-            .then((response) => response.data)
-        : [],
-
-    enabled: !!counterpartId,
-  });
+  const { api } = useMoniteContext();
+
+  return api.counterparts.patchCounterpartsIdAddressesId.useMutation(
+    undefined,
+    {
+      onSuccess: (counterpart) => {
+        toast.success(t(i18n)`Address “${counterpart.line1}” was updated.`);
+      },
+
+      onError: () => {
+        toast.error(t(i18n)`Failed to update Address.`);
+      },
+    }
+  );
 };
 
-export const useCreateCounterpartBank = (counterpartId: string) => {
+export const useCreateCounterpartBank = () => {
   const { i18n } = useLingui();
-  const { monite } = useMoniteContext();
-  const { add } = useCounterpartBankListCache(counterpartId);
-
-  return useMutation<
-    CounterpartBankAccountResponse,
-    Error,
-    CreateCounterpartBankAccount
-  >({
-    mutationFn: (bank) =>
-      monite.api.counterparts.createBankAccount(counterpartId, bank),
-
-    onSuccess: (bank) => {
-      add(bank);
-
-      toast.success(t(i18n)`Bank Account “${bank.name}” was created.`);
-    },
-
-    onError: () => {
-      toast.error(t(i18n)`Failed to create Bank Account.`);
-    },
-  });
+  const { api } = useMoniteContext();
+
+  return api.counterparts.postCounterpartsIdBankAccounts.useMutation(
+    undefined,
+    {
+      onSuccess: (bank) => {
+        toast.success(t(i18n)`Bank Account “${bank.name}” was created.`);
```

Comment:
How do you think, will the invalidation affect something?
```ts
await api.counterparts.getCounterpartsIdBankAccounts.invalidateQueries(
  {
    parameters: { path: { counterpart_id: counterpartId } },
  },
  queryClient
);
```

It did not exist in the original code, but what if we add it in advance?